python/revng/model/migrations/versions/v7.py

```diff
@@ -0,0 +1,10 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from revng.model.migrations import MigrationBase
+
+
+class Migration(MigrationBase):
+    def migrate(self, model):
+        pass
```

Comment:
We can't just leave this like this.

We need to properly define the current semantics of `Binaries`.

AFAIU, as of now, having none is allowed but the pypeline will fail at `Lift` time, correct?

Then, once we ditch the old pipeline, this will become mandatory and then we'll have to decided we want to do with the migration.

We need to document this somewhere, probably in the doc of `Binaries`.

Reply:
> AFAIU, as of now, having none is allowed but the pypeline will fail at Lift time, correct?

Not in this PR, but in the next PR where I'm implementing the first half of the pipeline the `Lift` pipe will check that segments map to a binary (in `checkPrecondition`)

> Then, once we ditch the old pipeline, this will become mandatory and then we'll have to decided we want to do with the migration.

Yes

> We need to document this somewhere, probably in the doc of Binaries.

So, something along the lines of "A model is valid without any `Binaries`, however do note that doing so will result in any pipe that needs to read the input binary to throw an error" in the docs?